From e9af2f54a4508f8200985f4587fee7b051198507 Mon Sep 17 00:00:00 2001
From: RigoLigo <rigoligo03@gmail.com>
Date: Sat, 16 May 2026 19:38:22 +0800
Subject: [PATCH 17/29] eeschema: remove file:// scheme for sheet file names

Since 593a1f5, file paths chosen from file picker launched from grids are prepended with "file://". This breaks sheet path processing, raising false invaldi file name warnings on Windows stable and also breaks nightly. This commit removed "file://" scheme prefix for sheet file names.
---
 eeschema/dialogs/dialog_sheet_properties.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/eeschema/dialogs/dialog_sheet_properties.cpp b/eeschema/dialogs/dialog_sheet_properties.cpp
index 827c1d76..38e0b028 100644
--- a/eeschema/dialogs/dialog_sheet_properties.cpp
+++ b/eeschema/dialogs/dialog_sheet_properties.cpp
@@ -285,6 +285,16 @@ bool DIALOG_SHEET_PROPERTIES::TransferDataFromWindow()
     // but unedited data from existing files can be bad.)
     sheetFileName = EnsureFileExtension( sheetFileName, FILEEXT::KiCadSchematicFileExtension );
 
+    // sheetFileName from the grid edit have URI scheme prepended (since 593a1f5).
+    // Remove the "file://" scheme, otherwise hierarchical sheet loading and checks below will all break
+    const wxString fileScheme = wxS( "file://" );
+    // Trim possible left side spaces
+    sheetFileName.Trim( false );
+    if( sheetFileName.StartsWith( fileScheme ) )
+    {
+        sheetFileName.Remove( 0, fileScheme.Len() );
+    }
+
     // Ensure sheetFileName is legal
     if( !IsFullFileNameValid( sheetFileName ) )
     {
-- 
2.43.0

